copy_from.go

```diff
@@ -105,9 +106,17 @@ func (ct *copyFrom) run(ctx context.Context) (int64, error) {
 	}
 	quotedColumnNames := cbuf.String()
 
-	sd, err := ct.conn.Prepare(ctx, "", fmt.Sprintf("select %s from %s", quotedColumnNames, quotedTableName))
-	if err != nil {
-		return 0, err
+	var oids []uint32
+	switch ct.mode {
+	case QueryExecModeCacheStatement, QueryExecModeCacheDescribe, QueryExecModeDescribeExec:
+		var err error
+		oids, err = ct.oidsFromStatementDescription(ctx, quotedColumnNames, quotedTableName)
+		if err != nil {
+			return 0, err
+		}
+	case QueryExecModeExec, QueryExecModeSimpleProtocol:
```

Comment:
I would have expected this to cause test failures, so maybe I'm missing something, but I don't see how this will work for `QueryExecModeExec` and `QueryExecModeSimpleProtocol`. `oids` will be `nil`.

Reply:
I need to pass `oids = nil` so that `buildCopyBuf` looks for the OIDs from the values. 

Here's more context https://github.com/jackc/pgx/pull/1412/files#r1045710050 I'll add some extra comments to avoid confusion.